src/StripeTests/Services/AutoPagingTest.cs

```diff
@@ -0,0 +1,105 @@
+namespace StripeTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Moq;
+    using Moq.Protected;
+    using Newtonsoft.Json;
+    using Stripe;
+    using Xunit;
+
+    public class AutoPagingTest : BaseStripeTest
+    {
+        [Fact]
+        public void ListAutoPaging()
+        {
+            // Set up stubbed requests
+            var response1 = new HttpResponseMessage(HttpStatusCode.OK);
+            response1.Content = new StringContent(GetResourceAsString("pageable_models.0.json"));
+            var response2 = new HttpResponseMessage(HttpStatusCode.OK);
+            response2.Content = new StringContent(GetResourceAsString("pageable_models.1.json"));
+            var response3 = new HttpResponseMessage(HttpStatusCode.OK);
+            response3.Content = new StringContent(GetResourceAsString("pageable_models.2.json"));
+
+            MockHandler.Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(m =>
+                        m.Method == HttpMethod.Get &&
+                        m.RequestUri.AbsolutePath == "/v1/pageablemodels"),
+                    ItExpr.IsAny<CancellationToken>())
+                .Returns(Task.FromResult(response1))
+                .Returns(Task.FromResult(response2))
+                .Returns(Task.FromResult(response3))
+                .Throws(new Exception("Unexpected invocation!"));
+
+            // Call auto-paging method
+            var service = new PageableService();
+            var options = new ListOptions()
+            {
+                Limit = 2,
+            };
+            var models = service.ListAutoPaging(options).ToList();
+
+            // Check results
+            Assert.Equal(5, models.Count);
+            Assert.Equal("pm_123", models[0].Id);
+            Assert.Equal("pm_124", models[1].Id);
+            Assert.Equal("pm_125", models[2].Id);
+            Assert.Equal("pm_126", models[3].Id);
+            Assert.Equal("pm_127", models[4].Id);
```

Comment:
So auto-paging basically builds the entire list in ram no matter what? You can't auto page while also getting items one by one and not filling your RAM with all your BTs history?

Reply:
You can. In these examples, I used `ToList()` to directly convert the iterator into a list, but you can also iterate over the objects one by one:

```c#
foreach (var balanceTransaction in balanceTransactionService.ListAutoPaging(listOptions))
{
  // Do something with balanceTransaction
}
```

Reply:
ah neat! Can you put a test for this in our test suite? I know people refer to the test suite a lot to find example. I guess this is where your `yield` applies. Fun how similar it is in each language you've shown me this approach!